src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,520 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+#:set ALL_KINDS_TYPES = INT_KINDS_TYPES + RC_KINDS_TYPES
+module stdlib_stats_distribution_uniform
+    use stdlib_kinds
+    use stdlib_error, only : error_stop
+    use stdlib_stats_distribution_PRNG, only : dist_rand
+
+    implicit none
+    private
+
+    real(dp), parameter  :: MESENNE_NUMBER = 1.0_dp / (2.0_dp ** 53 - 1.0_dp)
+    integer(int64), parameter :: INT_ONE = 1_int64
+
+    public :: uniform_distribution_rvs
+    public :: uniform_distribution_pdf
+    public :: uniform_distribution_cdf
```

Comment:
@Jim-215-Fisher @jvdp1 What do you think about renaming these to

```suggestion
    public :: rvs_uniform
    public :: pdf_uniform
    public :: cdf_uniform
```

So, more general name of the procedure first, and the specific distribution kind as a suffix. That makes more sense to me as I read these names left to right, but that's just me.

And I think we can omit the `distribution` name here, as it's obvious (IMO) from the module name. This makes it less likely that the user would rename them on import.

Reply:
Thanks, I may be now leaning even toward suggesting calling them just `rvs`, `pdf`, and `cdf`, for all distribution modules. So the user would do

```
use stdlib_stats_distribution_uniform, only: rvs, pdf, cdf
```

and if they use multiple distributions, they can do

```
use stdlib_stats_distribution_uniform, only: rvs_uni => rvs, pdf_uni => pdf, cdf_uni => cdf
use stdlib_stats_distribution_beta, only: rvs_beta => rvs, pdf_beta => pdf, cdf_beta => cdf
```

The downside is that if they do

```
use stdlib_stats_distribution_uniform
use stdlib_stats_distribution_beta
```

then they can't reference the conflicting names.

Otherwise I really like the concise names and that they'd be same across modules, and this may be even closer to the SciPy naming (except of course we can't namespace).

We could make a Discourse poll and see what names are most clear to people.

Reply:
@jvdp1 @awvwgk @ivan-pi what do you think about the naming for these functions?

Reply:
I'm fine with more verbose names. There is always the possibility to rename on import in case the names turn out to long.

Reply:
Sorry to sneak into this discussion, please feel free to ignore my comment if inappropriate.

As a very likely user of this great module (that I have test-driven a few months ago in my codes without issues), I would prefer to have `rvs_uniform` , `rvs_normal`, `rvs_beta`, etc... I agree that it would be a bit redundant to write
```
use stdlib_stats_distribution_uniform, only: rvs => rvs_uniform
```
but, as observed above, the "lazy", default mode of doing
```
use stdlib_stats_distribution_uniform
use stdlib_stats_distribution_normal
```
is safe from clashes in this way.

Just my 2c.

Reply:
Thanks, all, and @epagone your comment is very helpful. It looks like very short names (`rvs`, `pdf`, `cdf`) don't have support, and that we agree "`distribution`" is redundant. Then, we should decide between `rvs_uniform` and `uniform_rvs`. I think either works, and my preference for `rvs_uniform` is only slight. I'll search through the rest of the stdlib if either approach was more prevalent than the other in other modules.

Reply:
thank you @epagone for your comment. I agree with your suggestion: `*_uniform`, `*_normal`,...

Reply:
Here's a poll, maybe we'll learn something: https://fortran-lang.discourse.group/t/poll-naming-a-function-in-stdlib/1900.

Reply:
Based on the poll so far, 72% of respondents would be happy with `pdf_uniform` (prefers this or doesn't care), and 35% of respondents would be happy with `uniform_pdf` (prefers this or doesn't care). We can keep watching the poll, but I don't think this preference would change significantly.

Reply:
OK, I will change the naming according to the poll.